trailblazer/store/crud/update.py

```diff
@@ -41,167 +30,11 @@ def update_user_is_archived(self, user: User, archive: bool = True) -> None:
-    ) -> None:
-        """Update analysis jobs from supplied squeue results."""
-        if len(squeue_result.jobs) == 0:
-            return
-        self.delete_analysis_jobs(analysis)
-
-        session: Session = get_session()
-        for job in squeue_result.jobs:
-            job.step = reformat_squeue_result_job_step(
-                workflow=analysis.workflow, job_step=job.step
-            )
-            new_job = Job(
-                analysis_id=analysis.id,
-                slurm_id=job.id,
-                name=job.step,
-                status=job.status,
-                started_at=job.started_at,
-                elapsed=job.time_elapsed,
-            )
-            session.add(new_job)
-        session.commit()
-
-    def _update_analysis_from_slurm_squeue_output(
-        self, analysis: Analysis, analysis_host: str | None = False
-    ) -> None:
-        """Update analysis status based on current SLURM jobs status."""
-        slurm_job_id_file = Path(analysis.config_path)
-        queue_output = get_slurm_squeue_output(
-            analysis_host=analysis_host, slurm_job_id_file=slurm_job_id_file
-        )
-        squeue_result: SqueueResult = get_squeue_result(queue_output)
-        self.update_analysis_jobs_from_slurm_jobs(analysis=analysis, squeue_result=squeue_result)
-        LOG.debug(f"Status in SLURM: {analysis.case_id} - {analysis.id}")
-        LOG.debug(squeue_result.jobs)
-        analysis.progress = squeue_result.jobs_status_distribution.get(
-            SlurmJobStatus.COMPLETED, 0.0
-        )
-        analysis.status = get_current_analysis_status(squeue_result.jobs_status_distribution)
-        LOG.info(f"Updated status {analysis.case_id} - {analysis.id}: {analysis.status} ")
-        analysis.logged_at = datetime.now()
-        session: Session = get_session()
-        session.commit()
-
-    def update_tower_run_status(self, analysis_id: int) -> None:
-        """Query tower for entries related to given analysis, and update the Trailblazer database."""
-        analysis: Analysis | None = self.get_analysis_with_id(analysis_id)
-        tower_api: TowerAPIService | None = get_tower_api(
-            config_file_path=analysis.config_path, case_id=analysis.case_id
-        )
-
-        try:
-            self._update_analysis_from_tower_output(
-                analysis=analysis, analysis_id=analysis_id, tower_api=tower_api
-            )
-        except Exception as error:
-            LOG.error(f"Error logging case - {analysis.case_id} :  {type(error).__name__}")
-            analysis.status = TrailblazerStatus.ERROR
-            session: Session = get_session()
-            session.commit()
-
-    def _update_analysis_from_tower_output(
-        self, analysis: Analysis, analysis_id: int, tower_api: TowerAPIService
-    ):
-        LOG.info(f"Status in Tower: {analysis.case_id} - {analysis_id} - {tower_api.workflow_id}")
-        analysis.status = tower_api.status
```

Comment:
The refactored version calculates this status from the jobs persisted in the database for the analysis.

It might be the case that tower differs from slurm in how jobs are created and therefor this might be wrong.
If new jobs can be added when all the previous ones completed, the status will be incorrectly reported as completed.